sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaSensor.java

```diff
@@ -0,0 +1,254 @@
+        fileScan.command(arguments);
+        try {
+            fileScan.inheritIO();
+            Process process = fileScan.start();
+            int exitCode = process.waitFor();
+            if (exitCode == 0) {
+                String analyzedResultsFilePath = Paths.get(context.fileSystem().baseDir().getPath())
+                        .resolve(ISSUES_FILE_PATH)
+                        .toString();
+                String fileContent = getFileContent(analyzedResultsFilePath);
+                reportFileContent(context, pathAndInputFiles, fileContent);
+            } else {
+                logger.info("Unable to analyze ballerina file batch!");
+            }
+        } catch (IOException | InterruptedException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private String getFileContent(String analyzedResultsFilePath) {
+        String fileContent = "";
+        try (
+                FileReader fileReader = new FileReader(analyzedResultsFilePath, StandardCharsets.UTF_8);
+                BufferedReader bufferedReader = new BufferedReader(fileReader)
+        ) {
+            fileContent = bufferedReader.lines().collect(Collectors.joining());
+        } catch (IOException ex) {
+            logger.info("Unable to retrieve analysis results!");
+        }
+        return fileContent;
+    }
+
+    private void reportFileContent(SensorContext context,
+                                   Map<String, InputFile> pathAndInputFiles,
+                                   String fileContent) {
+        JsonArray balScanOutput = null;
+        try {
+            balScanOutput = JsonParser.parseString(fileContent).getAsJsonArray();
+        } catch (Exception ignored) {
+            logger.info("Unable to report analysis results!");
+        }
+
+        boolean reportingSuccessful = reportAnalysisIssues(context, balScanOutput, pathAndInputFiles);
+        if (reportingSuccessful) {
+            logger.info("Ballerina analysis successful!");
+        } else {
+            logger.info("Unable to analyze ballerina file batch!");
+        }
+    }
+
+    public boolean reportAnalysisIssues(SensorContext context, JsonArray analysisIssues,
+                                        Map<String, InputFile> pathAndInputFiles) {
+        if (analysisIssues == null) {
+            return false;
+        }
+        for (JsonElement issueElement : analysisIssues) {
+            JsonObject issue = issueElement.getAsJsonObject();
+            String absoluteFilePath = issue.get(ISSUE_FILE_PATH).getAsString();
+            InputFile inputFile = pathAndInputFiles.get(absoluteFilePath);
+            String issueType = issue.get(ISSUE_SOURCE).getAsString();
+            switch (issueType) {
+                case BUILT_IN -> reportIssue(inputFile, context, issue, true);
+                case EXTERNAL -> reportIssue(inputFile, context, issue, false);
+                default -> logger.info("Invalid Issue Format!");
```

Comment:
```suggestion
                default -> logger.info("Invalid issue format!");
```

Please recheck all logs.